web-ui/src/main/java/org/archcnl/ui/input/ruleeditor/ArchitectureRulesLayout.java

```diff
@@ -8,29 +8,32 @@
 import org.archcnl.domain.input.model.architecturerules.ArchitectureRule;
 import org.archcnl.ui.input.ButtonClickResponder;
 import org.archcnl.ui.input.CreateNewLayout;
+import org.archcnl.ui.input.InputView;
 import org.archcnl.ui.input.RulesOrMappingEditorView;
+import org.archcnl.ui.main.MainPresenter;
 
 public class ArchitectureRulesLayout extends RulesOrMappingEditorView
         implements PropertyChangeListener {
 
     private static final long serialVersionUID = 1L;
+    private InputView parent;
 
-    CreateNewLayout createNewRuleLayout =
-            new CreateNewLayout(
-                    "Architecture Rules",
-                    "Create new Arch Rule",
-                    new ButtonClickResponder() {
-                        @Override
-                        public void onButtonClick() {}
-                    });
     VerticalLayout rulesLayout = new VerticalLayout();
 
-    public ArchitectureRulesLayout() {
+    public ArchitectureRulesLayout(InputView Parent, MainPresenter mainPresenter) {
         RulesConceptsAndRelations.getInstance()
                 .getArchitectureRuleManager()
                 .addPropertyChangeListener(this);
         // Remove style property to makes no sense in this layout
         // TODO: Separate ArchitectureRulesLayout from CreateNewLayout
```

Comment:
These two comment lines belong to the line `createNewRuleLayout.getStyle().remove("border");` and thus should be left directly above that line.